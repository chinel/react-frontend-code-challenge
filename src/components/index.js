export { default as DivisionSummary } from "./DivisionSummary/DivisionSummary";
export { default as Header } from "./Header/Header";
export { default as InfoBox } from "./InfoBox/InfoBox";
export { default as ModuleHistory } from "./ModuleHistory/ModuleHistory";
export { default as TableSummary } from "/TableSummary/TableSummary";
