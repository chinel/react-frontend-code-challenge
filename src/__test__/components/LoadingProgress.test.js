import React from "react";
import { render } from "@testing-library/react";
import useStyles from "../../components/LoadingProgress/LoadingProgressStyles";
import { LoadingProgress } from "../../components";

function checkStyles() {
  let values;

  function MockedComponent() {
    values = useStyles();
    return null;
  }
  render(<MockedComponent />);
  return values;
}

describe("<LoadingProgress/>", () => {
  it("renders the <LoadingProgress/> with populated data", () => {
    const { container, getByTestId } = render(<LoadingProgress />);

    expect(getByTestId("loader")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("validate the return values of useStyles hook", () => {
    const validateStlyes = checkStyles();

    expect(validateStlyes.loading).toBeTruthy();
  });
});
