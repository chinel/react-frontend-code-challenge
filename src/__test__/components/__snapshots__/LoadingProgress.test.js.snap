// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingProgress/> renders the <LoadingProgress/> with populated data 1`] = `
<div
  style="display: flex; flex-direction: column; align-items: center; margin-top: 50px;"
>
  <div
    class="MuiCircularProgress-root makeStyles-loading-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    data-testid="loader"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
